src/main/java/filemanager/directorytracker/ScheduleFileDirectory.java

```diff
@@ -91,30 +61,39 @@ public void goThroughToCheckFile(Command command) {
                         .filter((file) -> Files.isRegularFile(file) && matchPattern(file.toString()))
                         .forEach(path -> {
                             try {
-                                readService.readJson(new FileInputStream(path.toString()), interactions);
+                                interactions.addAll(readService.readJson(new FileInputStream(path.toString())));
                             } catch (IOException e) {
                                 e.printStackTrace();
                             }
                         });
             } catch (IOException e) {
                 e.printStackTrace();
             }
-
             groupService.groupByClient(interactions, interactionsMap);
-            interactionsMap.forEach((clientName, interactionList) -> {
-                if (clientName.equalsIgnoreCase(command.getClient())) {
+            if (commandClientName == null || commandClientName.isEmpty()) {
+                interactionsMap.forEach((clientName, interactionList) -> {
                     String temporaryPath = getOutputPath() + clientName;
                     try {
                         writeService.writeXmlFile(interactionList, temporaryPath);
                     } catch (IOException e) {
                         e.printStackTrace();
                     }
-                }
-            });
+                });
+            } else {
+                interactionsMap.forEach((clientName, interactionList) -> {
+                    if (clientName.equalsIgnoreCase(commandClientName)) {
```

Comment:
Why do you need two identical if branches if you may just include its logic into this condition? Please merge them